docs/ref/models/fields.txt

```diff
@@ -1150,6 +1150,44 @@ when :attr:`~django.forms.Field.localize` is ``False`` or
     information on the difference between the two, see Python's documentation
     for the :mod:`decimal` module.
 
+``GeneratedField``
+------------------
+
+.. class:: GeneratedField(expression, output_field=None, db_persist=True, **kwargs)
+
+A field which is computed based on other fields in the model. Uses the database's
+``GENERATED ALWAYS AS`` syntax. Similar to a database view, this field is managed
+and updated by the database itself.
+
+.. attribute:: GeneratedField.expression
+
+    An :class:`Expression` used by the database to automatically set the field
+    value whenever the model is changed.
+
+    Several restrictions apply to generated columns, depending on the database.
+    A :class:`GeneratedField` should be immutable and only reference
+    concrete columns inside the model itself. References to other models' fields
+    is not supported. See your database's documentation for more information.
```

Comment:
This paragraph was mixing the concepts of columns and fields. Plus I think "deterministic" is the right word, rather than "immutable" (the input fields into the expression are allowed to mutate). My attempt to improve:

```suggestion
    The expressions should be deterministic and only reference fields within
    the model (in one database table). Database backends can impose further
    restrictions.
```

The "(in one database table)" is because a model can span multiple tables, in the case of multi-table inheritance.

Reply:
I pulled "immutable" from the Postgres docs. That said, "deterministic" works for me and I'm good with these changes.